code/game/objects/items/devices/laserpointer.dm

```diff
@@ -44,22 +52,78 @@
+		if(diode)
+			balloon_alert(user, "already has a diode!")
+			return
+		var/obj/item/stock_parts/attack_diode = attack_item
+		if(crystal_lens && attack_diode.rating < 3) //only tier 3 and up are small enough to fit
+			to_chat(user, span_warning("You try to jam \the [attack_item.name] in place, but \the [crystal_lens.name] is in the way!"))
+			playsound(src, 'sound/machines/airlock_alien_prying.ogg', 20)
+			if(do_after(user, 2 SECONDS, src))
+				var/atom/atom_to_teleport = pick(user, attack_item)
+				if(atom_to_teleport == user)
+					to_chat(user, span_warning("You jam \the [attack_item.name] in too hard and break \the [crystal_lens.name] inside, teleporting you away!"))
+					user.drop_all_held_items()
+				else if(atom_to_teleport == attack_item)
+					attack_item.forceMove(drop_location())
+					to_chat(user, span_warning("You jam \the [attack_item.name] in too hard and break \the [crystal_lens.name] inside, teleporting \the [attack_item.name] away!"))
+				do_teleport(atom_to_teleport, get_turf(src), crystal_lens.blink_range, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)
+				qdel(crystal_lens)
+			return
+		if(!user.transferItemToLoc(attack_item, src))
+			return
+		playsound(src, 'sound/items/screwdriver.ogg', 30)
+		diode = attack_item
+		balloon_alert(user, "installed \the [diode.name]")
+		//we have a diode now, try starting a charge sequence in case the pointer was charging when we took out the diode
+		recharging = TRUE
+		START_PROCESSING(SSobj, src)
+
+	if(istype(attack_item, /obj/item/stack/ore/bluespace_crystal))
+		if(crystal_lens)
+			balloon_alert(user, "already has a lens!")
+			return
+		//the crystal we're about to install
+		var/obj/item/stack/ore/bluespace_crystal/jammed_crystal = attack_item
+		if(diode && diode.rating < 3) //only lasers of tier 3 and up can house a lens
+			to_chat(user, span_warning("You try to jam \the [jammed_crystal.name] in front of the diode, but it's a bad fit!"))
+			playsound(src, 'sound/machines/airlock_alien_prying.ogg', 20)
+			if(do_after(user, 2 SECONDS, src))
+				var/atom/atom_to_teleport = pick(user, src)
+				if(atom_to_teleport == user)
+					to_chat(user, span_warning("You press on \the [jammed_crystal.name] too hard and are teleported away!"))
+					user.drop_all_held_items()
+				else if(atom_to_teleport == src)
+					forceMove(drop_location())
+					to_chat(user, span_warning("You press on \the [jammed_crystal.name] too hard and \the [src] is teleported away!"))
+				do_teleport(atom_to_teleport, get_turf(src), jammed_crystal.blink_range, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)
+				jammed_crystal.use_tool(src, user, amount = 1) //use only one if we were installing from a stack of crystals
+			return
+		if(!jammed_crystal.use_tool(src, user, amount = 1))
+			return
+		crystal_lens = new(src) //this could probably convert artificial crystals into pure ones, hmm
```

Comment:
i did not come up with a good solution for this then and i don't think i will come up with one now, pls halp

Reply:
feature, it teleports a normal crystal in. Bluespace

Reply:
oh #77092 could come in useful for solving this by making `fast_split_stack()` global, good to know